#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/CorrodedMetal.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.13358855831609984 -0.011887289046060881 -0.9909655843903769 3.5666461010498356
  position 1.9120437095798222 -0.3079688745960846 0.8496411552883403
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Shape {
  appearance CorrodedMetal {
  }
  geometry Cylinder {
    height 0.019
    radius 0.77
  }
}
Robot {
  translation 0 0 0.05
  rotation 0 0 1 1.7
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position 72.89059620222974
        anchor -0.05 0 0
      }
      device [
        PositionSensor {
          name "encoder_left"
        }
        RotationalMotor {
          name "motor_left"
        }
      ]
      endPoint Solid {
        translation -0.04999954940226955 6.810481509292587e-07 5.791955401881984e-07
        rotation 0.6891087850215734 -0.22587714114734386 0.6885554440371481 3.5862337560954805
        children [
          DEF wheel_direita Shape {
            appearance PBRAppearance {
              baseColor 0.960784 0.47451 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel_direita
        physics DEF wheel_physics Physics {
        }
      }
    }
    DEF Body Pose {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 1
          }
          geometry Cylinder {
            height 0.05
            radius 0.05
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 72.89059611285464
        anchor 0.05 0 0
      }
      device [
        PositionSensor {
          name "encoder_right"
        }
        RotationalMotor {
          name "motor_direita"
        }
      ]
      endPoint Solid {
        translation 0.050000449468013734 4.1701244895481237e-07 4.486904171413175e-07
        rotation 0.6891087903895501 -0.225877108590555 0.6885554493449468 3.5862336950520777
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.960784 0.47451 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.03
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
  ]
  name "MX0"
  boundingObject USE Body
  physics Physics {
  }
  controller "MXO_main"
}
Solid {
  children [
    DEF arena Shape {
      geometry Cylinder {
        height 0.019
        radius 0.77
      }
    }
  ]
  boundingObject USE arena
}
